### PR TITLE
[fix] KMeans ```score``` not properly checked in ```_onedal_*_supported``` and n_jobs support for ```score```

### PR DESCRIPTION
**Pull request by icfaust**
### Description

Fixes for issues in KMeans uncovered in #2054

- Add ```score``` to n_jobs checks
- Remove ```transform``` from n_jobs checks
- Fix ```_onedal_predict_supported``` to work with ```score```
- Move deprecation of sample_weights to before ```dispatch```, because ```dispatch``` will only pass arguments (not kwargs) to ```_onedal_cpu_supported``` and ```_onedal_gpu_supported``` (i.e. ```sample_weights``` has to be passed to ```dispatch``` as an argument) and if it is not None as an argument and not located on the same device as X, then it will fail.  The check must occur for the string beforehand, and convert it to None beforehand.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by md-shafiul-alam** [comment]
Overall looks good. Was there a bug other than "score" method not being supported with n_jobs?

**Comment by icfaust** [comment]
> Overall looks good. Was there a bug other than "score" method not being supported with n_jobs?

Good question, there are a number of score estimator attributes which aren't covered (and are collected together in #2054).  There was also something of note here, the ```transform``` attribute to kmeans is getting dropped from n_jobs support since it uses sklearn in the background:  https://github.com/intel/scikit-learn-intelex/blob/main/onedal/cluster/kmeans.py#L401

**Comment by icfaust** [comment]
Note: this PR is not to be merged until private CI issues are fixed. It will need to be re-validated with a merge of main.

**Review by samir-nasibli — COMMENTED** [review]
@icfaust I am good with this, but does it make sense first start with #2048?

File: sklearnex/cluster/k_means.py [hunk 1]
```diff
@@ -41,7 +41,12 @@
     from .._device_offload import dispatch, wrap_output_data
     from .._utils import PatchingConditionsChain
 
-    @control_n_jobs(decorated_methods=["fit", "predict", "transform", "fit_transform"])
+    if sklearn_check_version("1.6"):
+        from sklearn.utils.validation import validate_data
+    elif sklearn_check_version("1.0"):
+        validate_data = sklearn_KMeans._validate_data
+
```
Comment (Alexsandruss) [inline sklearnex/cluster/k_means.py:48]: Closed `if-else` without branch with undefined validate_data is preferred since sklearn<1.0 support is dropped.

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

File: sklearnex/cluster/k_means.py (hunk 1 above)
Comment (icfaust) [reply sklearnex/cluster/k_means.py]: @Alexsandruss done!

**Comment by icfaust** [comment]
> @icfaust I am good with this, but does it make sense first start with #2048?

Good question, I split it off due to the unrelated changes necessary for ```onedal_predict_supported``` to get it to properly work.  I guess #2048 can go first, but I was hoping to make sure to fix this bug quickly.

**Review by samir-nasibli — APPROVED** [review]
Looks good 👍 
Please just rebase your branch and run the CI once again

**Review by Alexsandruss — APPROVED** [review]
